src/test/java/com/xceptance/neodymium/tests/NeodymiumTest.java

```diff
@@ -52,29 +54,106 @@ public static void deleteTempFile(File tempFile)
     }
 
     public void check(Result result, boolean expectedSuccessful, int expectedRunCount, int expectedIgnoreCount, int expectedFailCount,
-                      String expectedFailureMessage)
+                      Map<String, String> expectedFailureMessages)
     {
-        Assert.assertEquals("Test successful", expectedSuccessful, result.wasSuccessful());
-        Assert.assertEquals("Method run count", expectedRunCount, result.getRunCount());
-        Assert.assertEquals("Method ignore count", expectedIgnoreCount, result.getIgnoreCount());
-        Assert.assertEquals("Method fail count", expectedFailCount, result.getFailureCount());
-
-        if (expectedFailureMessage != null)
+        String stackTrace = "";
+        for (Failure failue : result.getFailures())
+        {
+            stackTrace += failue.getTrace();
+        }
+        try
+        {
+            Assert.assertEquals("Test successful", expectedSuccessful, result.wasSuccessful());
+            Assert.assertEquals("Method run count", expectedRunCount, result.getRunCount());
+            Assert.assertEquals("Method ignore count", expectedIgnoreCount, result.getIgnoreCount());
+            Assert.assertEquals("Method fail count", expectedFailCount, result.getFailureCount());
+            for (int i = 0; expectedFailureMessages != null && i < result.getFailures().size(); i++)
```

Comment:
Put loop into `if` block that checks `expectedFailureMessages` for being not null.

I also noticed that `result.getFailureCount()` is equal to `result.getFailures().size()`. Thus, the failure count can be declared as `final` local var.

Reply:
Done